dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java

```diff
@@ -119,19 +121,17 @@ private List<Predicate> processQuery(String sortCol, String sortOrder, CriteriaB
                             HttpStatus.SC_BAD_REQUEST);
 
                 }
-                sortPath = version.get(sortCol);
-            }
-            if ("desc".equalsIgnoreCase(sortOrder)) {
-                query.orderBy(cb.desc(sortPath), cb.desc(versionId));
-            } else {
-                query.orderBy(cb.asc(sortPath), cb.asc(versionId));
+                sortExpression = version.get(sortCol);
             }
         } else {
-            if ("desc".equalsIgnoreCase(sortOrder)) {
-                query.orderBy(cb.desc(versionId));
-            } else {
-                query.orderBy(cb.asc(versionId));
-            }
+            sortExpression = cb.selectCase()
+                .when(cb.equal(versionId, representativeVersionId), 1)
+                .otherwise(0);
+        }
+        if ("desc".equalsIgnoreCase(sortOrder)) {
+            query.orderBy(cb.desc(sortExpression), cb.desc(lastModified), cb.desc(versionId));
```

Comment:
i.e. can you sort by whether or not a version is the representative version before sortExpression, last modified, versionID and/or what are the trade-offs

Reply:
These are the primary, secondary, and tertiary sort criteria.  That is, if `sortExpression` results in a tie, `lastModified` is used to break the tie, and is `lastModified` results in a tie, `versionId` is used to break the tie.